src/main/java/uk/gov/hmcts/ccd/config/JacksonUtils.java

```diff
@@ -117,4 +121,67 @@ private static JsonNode merge(JsonNode mainNode, JsonNode updateNode) {
         }
         return mainNode;
     }
+
+    public static String getValueFromPath(final String path, final Map<String, JsonNode> dataMap) {
+        final String[] jsonValue = new String[1];
+        dataMap.entrySet().forEach(entry -> {
+            if (path.contains(".")) {
+                String key = path.substring(0, path.indexOf("."));
+                String truncatedKey = path.substring(path.indexOf(".") + 1);
+
+                if (entry.getKey().equals(key)) {
+                    jsonValue[0] = findValueFromTruncatedKey(entry, truncatedKey);
+                }
+            } else if (entry.getKey().equals(path)) {
+                jsonValue[0] = getValue(entry.getValue());
+            }
+        });
+
+        return jsonValue[0];
+    }
+
+    private static String findValueFromTruncatedKey(Map.Entry<String, JsonNode> entry, String truncatedKey) {
+        if (entry.getValue().isArray()) {
+            return getValueFromArray(entry, truncatedKey);
+        } else {
+            JsonNode foundJsonNode = getNestedCaseFieldByPath(entry.getValue(), truncatedKey);
+            if (foundJsonNode != null) {
+                return foundJsonNode.textValue();
+            }
+        }
+        return null;
+    }
+
+    private static String getValueFromArray(Map.Entry<String, JsonNode> entry, String truncatedKey) {
+        ArrayNode arrayNode = ((ArrayNode)entry.getValue());
+        final var arrayIndex = truncatedKey.substring(0, truncatedKey.indexOf("."));
+        final var keyToArrayContent = truncatedKey.substring(truncatedKey.indexOf(".") + 1);
+        if (StringUtils.isNumeric(arrayIndex)) {
+            JsonNode jsonNodeFromArray = arrayNode.get(Integer.parseInt(arrayIndex));
+            JsonNode nestedCaseFieldByPath =
+                getNestedCaseFieldByPath(jsonNodeFromArray, keyToArrayContent);
+            if (nestedCaseFieldByPath == null) {
+                nestedCaseFieldByPath = jsonNodeFromArray.get("value");
+            }
+            return getValue(nestedCaseFieldByPath);
+        }
+        return null;
+    }
+
+    private static String getValue(JsonNode jsonNode) {
+        String returnValue = null;
+        if (jsonNode != null) {
+            if (jsonNode instanceof IntNode) {
+                returnValue = jsonNode.toString();
+            } else {
+                if (jsonNode.iterator().hasNext()) {
+                    returnValue = jsonNode.iterator().next().textValue();
+                }
+                if (returnValue == null) {
+                    returnValue = jsonNode.textValue();
+                }
```

Comment:
Will the below suffice so that the double `if` go away?
returnValue =  jsonNode.iterator().hasNext() ? jsonNode.iterator().next().textValue() : jsonNode.textValue();

it's assumed that when `jsonNode.iterator().hasNext()` is true, `jsonNode.iterator().next().textValue()` will not be null

Reply:
Done [c129a9a](https://github.com/hmcts/ccd-data-store-api/pull/1872/commits/c129a9a6a6fdff4226e65e0b027691d3cd49352f)